src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -214,35 +214,46 @@ void handle_reshape::run(program& p) {
-                if (!reshape_layout.compatible(target_layout)) {
+
+                // Check whether input reorder is required for format change
+                if (input_layout.format != format::get_default_format(input_layout.get_rank())) {
                     auto reshape_input = std::make_shared<reorder>("reorder:_reshape_input_" + node->id(),
-                                                                   cldnn::input_info(input_node.id(), input_port),
-                                                                   target_format,
-                                                                   reshape_layout.data_type);
+                                                                cldnn::input_info(input_node.id(), input_port),
+                                                                target_format,
+                                                                reshape_layout.data_type);
                     GPU_DEBUG_LOG << "reshape_handler: " << reshape_input->id
                         << " input_info : " << reshape_input->dependencies().front().to_string() << std::endl;
                     auto& reshape_input_node = p.get_or_create(reshape_input);
                     p.add_intermediate(reshape_input_node, *node, 0, reshape_input_node.get_dependencies().empty());
                     reshape_input_node.recalc_output_layout();
+                }
 
+                // Check whether output reorder is required for format change
+                if (reshape_layout.format != target_format) {
                     auto reshape_users = node->get_users();
                     for (const auto& user : reshape_users) {
                         auto reshape_output = std::make_shared<reorder>("reorder:_reshape_output_" + node->id(),
-                                                                        user->id(),
+                                                                        node->id(),
                                                                         reshape_layout.format,
                                                                         reshape_layout.data_type);
+                        GPU_DEBUG_LOG << "reshape_handler: " << reshape_output->id
+                            << " input_info : " << reshape_output->dependencies().front().to_string() << std::endl;
                         auto& reshape_output_node = p.get_or_create(reshape_output);
                         p.add_intermediate(reshape_output_node,
-                                           *user,
-                                           *node,
-                                           reshape_output_node.get_dependencies().empty());
+                                        *user,
+                                        *node,
+                                        reshape_output_node.get_dependencies().empty());
                         reshape_output_node.recalc_output_layout();
```

Comment:
Random spot:
I am curious: isnt it better not to split reorder at all if the target condition is not met (at line 104), instead of adding reorders before and after..?

Reply:
This part is not related with "reorder_node_to_split". There is no reorder between the reshape and preds/users in this case. This is all about format handling in reorder and shape handling in reshape.